src/App.js

```diff
@@ -1,17 +1,130 @@
-import './App.css';
+/* eslint-disable no-console */
+/* eslint-disable react/jsx-props-no-multi-spaces */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable max-classes-per-file */
 import React from 'react';
+import { Dropdown, Option } from './Dropdown';
 
-function App() {
-  // fetches JSON data passed in by flask.render_template and loaded
-  // in public/index.html in the script with id "data"
-  // const args = JSON.parse(document.getElementById("data").text);
-
-  // This main page will be further designed with react components
-  // so the user can interact with the main app page. Some Data will also
-  // be present for viewing when user lands on this page.
-  return (
-    <h1> ! Welcome To Main App Page User !</h1>
-  );
+// This component will handle the diet lookup feature which will direct user to
+// the best overall Diets.
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: '' };
+
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSelect(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log(this.state.value);
+  }
+
+  render() {
+    return (
+      <div>
+        <h1> Find Out Best Overall Diets</h1>
+        <Dropdown
+          buttonText="Submit"
+          onChange={this.handleSelect}
+          action="https://health.usnews.com/best-diet/best-diets-overall"
+        >
+          <Option value="Click to see options" />
+          <Option value="Mediterranean Diet" />
+          <Option value="DASH Diet" />
+          <Option value="The Flexitarian Diet" />
+          <Option value="Weight Watchers Diet" />
+          <Option value="Mayo Clinic Diet" />
+          <Option value="The MIND Diet" />
+        </Dropdown>
+        <p>
+          You selected
+          {' '}
+          {this.state.value}
+          {' '}
+        </p>
+      </div>
+    );
+  }
+}
+// This component will handle the search function for the main app page. This search
+// function will take a users input and use the USDA API to get data back from a certain
+// food.
+class Search extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: null,
+      peoples: [],
+    };
+  }
+
+  componentDidMount() {
+    this.serachPeople(this.state.query);
+  }
+
+  onChange(e) {
+    this.setState({ query: e.target.value }, () => {
+      if (this.state.query && this.state.query.length > 1) {
+        if (this.state.query.length % 2 === 0) {
+          this.serachPeople(this.state.query);
+        }
+      } else {
+        this.serachPeople(this.state.query);
+      }
+    });
+  }
+
+  serachPeople(query) {
+    const apiKey = process.env.REACT_APP_API_KEY;
```

Comment:
I don't think `process` is available on the frontend. We should use the route method I mentioned on the last PR.

Reply:
What do you mean by this statement? The process is my attempt to be able to pass my API Key to the fetch react API and the USDA endpoint. I am not sure if I did this correctly or not to be able to get response data back.